### PR TITLE
Nimblfmp 569

### PR DESCRIPTION
**Pull request by Akayeshmantha**

### CODE REVIEW

File: catalogue-service-micro/src/main/java/eu/nimble/service/catalogue/impl/CatalogueController.java [hunk 1]
```diff
@@ -737,6 +737,33 @@ public ResponseEntity getCatalogueInSemanticFormat(@ApiParam(value = "uuid of th
         }
     }
 
+    @RequestMapping(value = "/catalogue/{partyId}",
+            produces = {"application/json"},
+            method = RequestMethod.GET)
+    public ResponseEntity getCatalogues(@ApiParam(value = "Identifier of the party for which the catalogue to be retrieved", required = true) @PathVariable String partyId,
```
Comment (dogukan10) [inline catalogue-service-micro/src/main/java/eu/nimble/service/catalogue/impl/CatalogueController.java]: The name of the service is a bit misleading. Could you please rename it as getCatalogueIds or something like that ?

Comment (suatgonul) [reply catalogue-service-micro/src/main/java/eu/nimble/service/catalogue/impl/CatalogueController.java]: Let's have it more self-descriptive: `getAllCatalogueIdsForParty`

Comment (Akayeshmantha) [reply catalogue-service-micro/src/main/java/eu/nimble/service/catalogue/impl/CatalogueController.java]: Sure guys.

File: catalogue-service-micro/src/main/java/eu/nimble/service/catalogue/impl/CatalogueController.java [hunk 2]
```diff
@@ -737,6 +737,33 @@ public ResponseEntity getCatalogueInSemanticFormat(@ApiParam(value = "uuid of th
...
+        } catch (Exception e) {
+            return createErrorResponseEntity("Failed to get catalogues for party id: " + partyId, HttpStatus.INTERNAL_SERVER_ERROR, e);
+        }
+
+        if (catalogueIds == null) {
+            log.info("No default catalogue for party: {}", partyId);
+            return ResponseEntity.status(HttpStatus.NOT_FOUND).body(String.format("No default catalogue for party: %s", partyId));
+        }
+
+        log.info("Completed request to get default catalogue for party: {}", partyId);
```
[the remark this answers is not in the document]
Comment (Akayeshmantha) [reply catalogue-service-micro/src/main/java/eu/nimble/service/catalogue/impl/CatalogueController.java]: Sure @dogukan10 .

**Pull request merged**